src/Sarif.UnitTests/Visitors/InsertOptionalDataVisitorTests.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT        
+// license. See LICENSE file in the project root for full license information. 
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using FluentAssertions;
+using Microsoft.CodeAnalysis.Sarif.Readers;
+using Microsoft.CodeAnalysis.Sarif.UnitTests;
+using Newtonsoft.Json;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.Sarif.Visitors
+{
+    public class InsertOptionalDataVisitorTests : FileDiffingTests
+    {
+        // To rebaseline all test files set this value to true and rerun the testa
+        private static bool s_rebaseline = false;
+
+        private ITestOutputHelper _outputHelper;
+
+        private static string GetTestDirectory(string subdirectory = "")
+        {
+            return Path.GetFullPath(Path.Combine(@".\TestData", subdirectory));
+        }
+
+        private static string GetProductTestDataDirectory(string subdirectory = "")
+        {
+            return Path.GetFullPath(Path.Combine(@"..\..\..\..\..\\src\Sarif.UnitTests\TestData", subdirectory));
```

Comment:
>..\..\..\..\..\\ [](start = 51, length = 16)

This is another reason you might fail on AppVeyor. You're assuming that the directory from which the tests are run is next door to the `src` directory.

Do you really need to pull anything from the `src` directory? Isn't the project file already copying the files you need to the output directory? (If it isn't, it could.)

Reply:
This helper is used to write data back to the source directory, to refresh baselines. Should only be used in an ad hoc rebaseline scenario on the dev box (not in appveyor). Closing by 'non-obvious design'. Will decorate the helper with a comment.

---
In reply to: [205981836](https://github.com/Microsoft/sarif-sdk/pull/974#discussion_r205981836) [](ancestors = 205981836)